tests/components/myuplink/test_switch.py

```diff
@@ -75,3 +77,19 @@ async def test_switch_off(
     )
     await hass.async_block_till_done()
     mock_myuplink_client.async_set_device_points.assert_called_once()
+
+
+async def test_api_failure(
+    hass: HomeAssistant,
+    mock_myuplink_client: MagicMock,
+    setup_platform: None,
+) -> None:
+    """Test handling of exception from API."""
+
+    with pytest.raises(HomeAssistantError):
+        mock_myuplink_client.async_set_device_points.side_effect = ClientError
+        await hass.services.async_call(
+            TEST_PLATFORM, SERVICE_TURN_OFF, {ATTR_ENTITY_ID: ENTITY_ID}, blocking=True
```

Comment:
Please add a parameter for the service name so we can test both turn on and turn off. Use `pytest.mark.parametrize`.

Here's an example with two parameters.

https://github.com/home-assistant/core/blob/6e91776d6561a1707842462d4b77cfad28e28ea8/tests/components/sharkiq/test_vacuum.py#L177-L185